docs/blog/2018-10-26-introducing-gatsby-themes/index.md

```diff
@@ -0,0 +1,154 @@
+
+# Why Themes?
+
+Defining themes as the composition of Gatsby sites allows us to start
+solving a variety of use cases. For example, when a site gets built as
+part of a wider product offering it's often the case that one team
+will build out a set of functionality, including branding elements,
+and the other teams will mostly consume this functionality. Themes
+allow us to distribute this functionality as an npm package and allow
+the customization of various branding elements through our
+gatsby-config.
+
+# Mechanics of Theming
+
+At a base level, theming simply combines the gatsby-config of the
+theme with the gatsby-config of your site. Since it's an experimental
+feature, we use an experimental namespace to declare themes in
+gatsby-config.
+
+```js
+module.exports = {
+  __experimentalThemes: [["gatsby-theme-blog", {}]],
+}
+```
+
+Themes often need to be parameterized for various reasons, such as
+changing the base url for subsections of a site or applying branding
+variables. We can do this through the theme options if we define our
+theme's gatsby-config as a function that returns an object.
+
+```js
+module.exports = {
+  __experimentalThemes: [
+    [
+      "gatsby-theme-blog",
+      {
+        some: "value",
+      },
+    ],
+  ],
+}
+```
+
+Themes also function as plugins and any config passed into the theme
+in your gatsby-config will also be passed to your theme's gatsby-\*
+files as plugin options. This allows themes to override any settings
+inherited from the theme's own plugin declarations or apply gatsby
+lifecycle hooks such as on-create-page.
+
+# Next Steps
+
+## Sub Themes and Overriding
+
+This is just the first step and it enables us to experiment with
+further improvements in userland before merging them into
+core. Sub-theming, for example, is a critical part of a theming
+ecosystem that is currently missing from core. Overriding theme
+elements is possible on a coarse level right now in userland. If, for
+example, a theme defines a set of pages using createPage we can define
+a helper function that will look for the page component first in the
+user's site and then fall back to the theme's default implementation.
+
+```js
+const withThemePath = relativePath => {
+  let pathResolvedPath = path.resolve(relativePath)
+  let finalPath = pathResolvedPath
+
+  try {
+    // check if the user's site has the file
+    require.resolve(pathResolvedPath)
+  } catch (e) {
+    // if the user hasn't implemented the file,
+    finalPath = require.resolve(relativePath)
+  }
+
+  return finalPath
+}
+```
+
+Then in our theme's createPage call, we simply use the helper to let
+the user optionally override the default component.
+
+```
+createPage({
+  path: post.node.fields.slug,
+  component: withThemePath('./src/templates/blog-post.js')
+})
+```
+
+This doesn't allow us to make more granular overrides of different
+components, but it does allow us to replace the rendering of pages and
+other whole elements.
```

Comment:
Is the intention of this to set up something that will be fixed soon? It's not entirely clear if this section is saying, "You should do this," or, "This is the only way to do things _now_, but we're going to fix it shortly."

Reply:
Oh, that's nice! Tag me on that PR, please, because I'd love to dig through it. I have some initial thoughts that I'll keep to myself to not start bikeshedding in a PR about something else. 😄